library/ssl_cli.c

```diff
@@ -840,10 +840,11 @@ static int ssl_validate_ciphersuite(
         return( 1 );
 #endif
 
-    /* Don't suggest PSK-based ciphersuite if no PSK is available. */
+    /* Don't suggest PSK-based ciphersuite if no PSK is available
+     * and if no PSK identity hint callback is installed. */
 #if defined(MBEDTLS_KEY_EXCHANGE_SOME_PSK_ENABLED)
     if( mbedtls_ssl_ciphersuite_uses_psk( suite_info ) &&
-        ssl_conf_has_static_psk( ssl->conf ) == 0 )
+        ssl_conf_has_static_psk( ssl->conf ) == 0 && ssl->conf->f_psk == NULL )
```

Comment:
The new condition should be in a new line.

Reply:
solved